switching/input/messages/R1.py

```diff
@@ -174,15 +174,31 @@ def contracte(self):
     #         data.append(C1.PuntMesura(i))
     #     return data
 
-    def get_subtypes(self, type):
+    def get_subtypes(self, type=None):
+        if not type:
+            type = self.sollicitud.tipus
         return [x['code'] for x in SUBTYPES_R101 if x['type'] == type]
```

Comment:
`type` is a reserved word try to change for other name.

I don't like mixing functions: static functions and instance methods. Maybe the best practice is to have a module level function:
```python
def get_subtypes(r1_type):
    return [x['code'] for x in SUBTYPES_R101 if x['type'] == type]
```
Then in the class:
```python
class R1:
    def get_suptypes(self):
        return get_subtypes(self.sollicitud.tipus)
```
Then if you want all the subtypes from a type of R1 is not necessary to have an R1 message instantiated and only calling the module function `get_subtypes()`.
```python
from switching.input.messages.R1 import get_subtypes
get_subtypes('01')
```